tyk-headless/templates/deployment-gw-repset.yaml

```diff
@@ -86,13 +86,22 @@ spec:
           - name: TYK_GW_STORAGE_ENABLECLUSTER
             value: "{{ default "false" .Values.redis.enableCluster }}"
           - name: TYK_GW_STORAGE_PASSWORD
-            value: "{{ .Values.redis.pass }}"
+            valueFrom:
+              secretKeyRef:
+                name: secrets
```

Comment:
I suggest making it possible to optionally specify a secret name as a value in case user doesn't want to store plaintext secrets in the Helm values and would rather provide the k8s `Secret` externally.

Same in the other charts.

Reply:
@excieve Could you give some examples of what is needed here? Thanks!

Reply:
@excieve I have added option for user to provide the k8s `Secret` externally. Please review it when you have a moment. Thanks!